packages/url-state-provider/src/index.ts

```diff
@@ -6,23 +6,36 @@
 import LZString from "lz-string"
 import superstate from "./superstate"
 
+// Define the options type
+type EncodeOptions = {
+  mode?: "humanize" | "auto"
+}
+type UpdateStateOptions = {
+  merge?: boolean
+}
+type HistoryOptions = {
+  state?: boolean
+  title?: string
+  replace?: boolean
+}
+
 const jsonURLSerializer = superstate()
 const SEARCH_KEY = "__s"
 
 const regex = new RegExp(SEARCH_KEY + "=([^&]+)")
 /**
  * Variable where to host listeners for history changes
  */
-var onHistoryChangeListeners = {}
-var onGlobalChangeListeners = []
+const onHistoryChangeListeners: { [key: string]: any } = {}
+const onGlobalChangeListeners: any[] = []
 
 /**
  * Encode json data using json-url or lz-string. It automatically detects the best encoding.
  * @param {JSON} json data to be encoded
  * @param {Object} options options for the encoding. Possible values: mode: "auto" or "humanize"
  * @returns encoded string
  */
-function encode(json, options = {}) {
+function encode(json: object, options: EncodeOptions = {}) {
```

Comment:
Wouldn’t it be better to make the options parameter optional and define a return value for encode?

```typescript
function encode(json: object, options?: EncodeOptions): string {
  // Set options to an empty object if it's undefined
  options = options || {};
  // ...
```

This way, you can call encode without passing options, allowing the tests to remain unchanged.